components/collection/HeroButtonEditCollection.vue

```diff
@@ -34,47 +35,83 @@ const { transaction, status, isLoading } = useTransaction()
 const { $i18n } = useNuxtApp()
 const { urlPrefix } = usePrefix()
 const route = useRoute()
-
+const collectionId = route.params.id.toString()
+const collectionPermissionSettings = ref<CollectionMintSetting>()
 const isModalActive = ref(false)
+const editedCollection = ref<UpdateCollection>()
 
 const collectionMetadata = computed(() =>
-  props.collection
+  props.collection && collectionPermissionSettings.value
     ? {
         name: props.collection.meta.name,
         description: props.collection.meta.description,
         image: props.collection.meta.image,
         imageType: props.collection.meta.type,
         banner: props.collection.meta.banner || undefined,
         max: props.collection.max,
+        mintingSettings: collectionPermissionSettings.value!,
       } as CollectionEditMetadata
     : null)
 
 const updateMetadata = (a: UpdateCollection, b: UpdateCollection) => {
   const getMetadataKey = (m: UpdateCollection) => {
-    const { max, ...rest } = m
+    const { max, mintingSettings, ...rest } = m
     return JSON.stringify(rest)
   }
 
   return getMetadataKey(a) !== getMetadataKey(b)
 }
 
-const editCollection = async (collection: UpdateCollection) => {
+const shouldUpdatePermission = (a: CollectionMintSetting, b: CollectionMintSetting) => {
+  return a.price !== b.price || a.mintType !== b.mintType || a.holderOf !== b.holderOf
+}
+
+const editCollection = async (updatedCollection?: UpdateCollection) => {
   isModalActive.value = false
 
+  const collection = updatedCollection || editedCollection.value!
+
+  // retry action
+  if (updatedCollection) {
+    editedCollection.value = updatedCollection
+  }
+
   if (!collectionMetadata.value) {
     return
   }
 
   await transaction({
     interaction: Collections.UPDATE_COLLECTION,
-    collectionId: route.params.id.toString(),
+    collectionId: collectionId,
     collection,
     update: {
       metadata: updateMetadata(collection, collectionMetadata.value),
       max: collection.max !== collectionMetadata.value.max,
+      permission: shouldUpdatePermission(collection.mintingSettings, collectionPermissionSettings.value!),
     },
     urlPrefix: urlPrefix.value,
     successMessage: $i18n.t('edit.collection.success'),
   })
 }
+
+watch(computed(() => collectionId), async () => {
+  const { apiInstance } = useApi()
+  const api = await apiInstance.value
+  const config = await api.query.nfts.collectionConfigOf(collectionId)
+  const mintSettings = (config.toHuman() as { mintSettings: {
+    price: string
+    mintType: CollectionMintSettingType
+    holderOf: string
+  } }).mintSettings
```

Comment:
make it a util `getCollectionMintSettings`

```suggestion
  const mintSettings = getCollectionMintSettings(collectionId)
```

Reply:
updated